--- a/toolchain/features/legacy/BUILD.bazel
+++ b/toolchain/features/legacy/BUILD.bazel
@@ -24,8 +24,12 @@
         "@rules_cc//cc/toolchains/actions:compile_actions",
     ],
-    args = [
-        # All warnings are enabled.
-        "-Wall",
+    args = select({
+        "@llvm//constraints/windows/abi:msvc": [
+            # clang-cl interprets -Wall as /Wall.
+            "/W3",
+        ],
+        "//conditions:default": ["-Wall"],
+    }) + [
         # Enable a few more warnings that aren't part of -Wall.
         "-Wthread-safety",
         "-Wself-assign",
