diff --git a/BUILD.bazel b/BUILD.bazel
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -4581,6 +4581,8 @@ cc_library(
   name = "simdutf",
   srcs = ["third_party/simdutf/simdutf.cpp"],
   hdrs = ["third_party/simdutf/simdutf.h"],
+  includes = ["third_party/simdutf"],
+  visibility = ["//visibility:public"],
   copts = select({
         "@v8//bazel/config:is_clang": ["-std=c++20"],
         "@v8//bazel/config:is_gcc": ["-std=gnu++2a"],
